akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardRegion.scala

```diff
@@ -218,7 +219,18 @@ object ShardRegion {
    */
   def gracefulShutdownInstance = GracefulShutdown
 
-  sealed trait ShardRegionQuery
+  /** Base marker trait for cluster sharding queries. */
+  private[akka] trait ShardingQuery
+  private[akka] trait ShardingQueryResponse
+
+  /** Broad response for cases where, e.g. the `ShardCoordinator` is restarting or
+   * in handover on a new node, cluster sharding initialization is in progress.
+   */
+  case object ClusterShardingInitializing extends ShardingQueryResponse
+
+  /** Base marker trait for shard region queries. */
+  sealed trait ShardRegionQuery extends ShardingQuery
+  sealed trait ShardRegionQueryResponse extends ShardingQueryResponse
```

Comment:
```
* Not for user extension
*/
@DoNotInherit sealed ...
```

(for Java users who doesn't have `sealed`)

Reply:
On both, or I think we'd want not just the region level but also the cluster level, WDYT?